src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -79,15 +79,16 @@ private bool ShouldAddNullableAnnotation(ITypeSymbol type)
             switch (type.NullableAnnotation)
             {
                 case CodeAnalysis.NullableAnnotation.Annotated:
-                    if (format.MiscellaneousOptions.IncludesOption(SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier) &&
+                    if ((format.MiscellaneousOptions.IncludesOption(SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier) ||
+                        format.MiscellaneousOptions.IncludesOption(SymbolDisplayMiscellaneousOptions.IncludeNonNullableReferenceTypeModifier)) &&
```

Comment:
Is this change needed? (It seems strange that `IncludeNonNullable...` implies `IncludeNullable...` but not the reverse.)

Reply:
I'd be okay with making the two flags independent, but I couldn't think of a scenario when we would want to print `!` annotations but not `?` annotations.

Reply:
@333fred What's your thought on the relationship of these two flags?

Previously, we asserted that we only use the `!` flag when we also use the `?` flag. But an assertion isn't sufficient for public APIs, obviously. We could make the flag independent, or we could throw an exception when an invalid combination of flags is produced.